src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java

```diff
@@ -0,0 +1,116 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import static org.junit.Assert.assertTrue;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.FindPatientPage;
+import org.openmrs.reference.page.RegistrationEditSectionPage;
+import org.openmrs.reference.page.RegistrationSummaryPage;
+import org.openmrs.uitestframework.test.TestData;
+
+public class EditDemographicsAndContactInfoSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+	private RegistrationEditSectionPage registrationEditSectionPage;
+	private RegistrationSummaryPage registrationSummaryPage;
+
+	@Before(RunTest.HOOK.SELENIUM_EDIT_DEMOGRAPHICS)
+	public void visitPatientDashboard() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+		findPatientPage = (FindPatientPage) homePage.goToFindPatientRecord()
+				.waitForPage();
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_EDIT_DEMOGRAPHICS)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Edit Registration Information from Patient dashboard")
+	public void loadRegistrationSummaryPage() {
+		registrationSummaryPage = (RegistrationSummaryPage) dashboardPage
+				.goToRegistrationSummary().waitForPage();
+	}
+
+	@Then("the system loads Registration Summary Page")
+	public void systemloadsRegistrationEditPage() {
+		assertTrue(registrationSummaryPage.containsText("DEMOGRAPHICS"));
+	}
+
+	@When("a user clicks on Edit link on Registration Summary Page")
+	public void loadDemographicsEditSection() {
+		registrationEditSectionPage = (RegistrationEditSectionPage) dashboardPage
+				.clickOnEditPatient().waitForPage();
+	}
+
+	@And("the system loads the edit demographics section")
+	public void systemloadsEditDemographicSection() {
+		assertTrue(textExists("Demographics"));
+	}
+
+	@When("a user enters the demographics of the patient")
+	public void captureNewDemographics() {
+		registrationEditSectionPage.enterGivenName("Jane");
+		registrationEditSectionPage.enterMiddleName("Adyeri");
+		registrationEditSectionPage.enterFamilyName("Namata");
+		registrationEditSectionPage.selectPatientGender("Male");
+		registrationEditSectionPage.clickOnBirthdateLabel();
+		registrationEditSectionPage.enterBirthDay("23");
+		registrationEditSectionPage.selectBirthMonth("May");
+		registrationEditSectionPage.enterBirthYear("1990");
+	}
+
+	@Then("the system saves the updated patient demographics")
+	public void systemSavesUpdateDemographics() {
+		assertTrue(textExists("DEMOGRAPHICS"));
```

Comment:
Does the text appear only after saving the demographics?

Reply:
@kdaud  after saving the Demographics the test goes back to the  Registration Summary Page when DEMOGRAPHICS is visiible

Reply:
which word is displayed when the demographics are saved in the db?

Reply:
I wanted to upload for you a screen shot but it can't, you could try it out locally on your side and see.

Reply:
Instead of asserting texts on the page, better to assert that the demographics table is not null since after saving the data. use assertNotNul().